### ISSUE TITLE
New generator based on the text contents

### ISSUE THREAD

**Issue by ymark** [isl/x3ml#90]
Request for implementing a new generator that will generate IDs based on the textual contents. Furtermore if there are similar texts it will re-use the generated IDs. For example given the following XML input
```
<entry>text a</entry>
<entry>text b</entry>
<entry>text a</entry>
```

It will generate only two IDs

**Comment by ymark** [comment]
The generator has been implemented.  The generator class is gr.forth.TextualContent. It receives as input only one parameter with name "text". Its functionality is the following: Given a particular String value it generates a URN (using the scheme URN:UUID:<UUID_GENERATED_VALUE>). The generator is responsible for producing different values for different input texts. As a result if a text value is given twice then only one URN will be generated (in the first time and re-used afterwards). An example follows: 

Generator policy:
```
<?xml version="1.0" encoding="UTF-8"?>
<generator_policy>
    <generator name="URNfromTextualContent">
        <custom generatorClass="gr.forth.TextualContent">
            <set-arg name="text" type="xpath"/>
        </custom>
    </generator>
    <generator name="URIorUUID">
        <custom generatorClass="gr.forth.URIorUUID">
            <set-arg name="text"/>
        </custom>
    </generator>
</generator_policy>
``` 

Input file:
```
<?xml version="1.0" encoding="UTF-8"?>
<dataroot>
    <COIN>
        <ID>http://localhost/coin_1</ID>
        <COUNTRY_ID>Ancient Greece</COUNTRY_ID>
    </COIN>
    <COIN>
        <ID>http://localhost/coin_2</ID>
        <COUNTRY_ID>Ancient Rome</COUNTRY_ID>
    </COIN>
    <COIN>
        <ID>http://localhost/coin_3</ID>
        <COUNTRY_ID>
		Ancient Greece
	</COUNTRY_ID>
    </COIN>
</dataroot>
```

X3ML Mappings file: 
```
<?xml version="1.0" encoding="UTF-8"?>
<x3ml version="1.0" source_type="xpath">
    <namespaces>
        <namespace prefix="loc" uri="http://localost/schema/"/>
        <namespace prefix="rdf" uri="http://www.w3.org/1999/02/22-rdf-syntax-ns#"/>
        <namespace prefix="rdfs" uri="http://www.w3.org/2000/01/rdf-schema#"/>
    </namespaces>
    <mappings>
        <mapping>
            <domain>
                <source_node>//COIN</source_node>
                <target_node>
                    <entity>
                        <type>loc:Man-Made_Object</type>
			<instance_generator name="URIorUUID">
				<arg name="text" type="xpath">ID/text()</arg>
			</instance_generator>
                    </entity>
                </target_node>
            </domain>
            <link>
                <path>
                    <source_relation><relation>COUNTRY_ID</relation></source_relation>
                    <target_relation>
                        <relationship>loc:has_country</relationship>
                    </target_relation>
                </path>
                <range>
                    <source_node>COUNTRY_ID</source_node>
                    <target_node>
                        <entity>
                            <type>loc:Country</type>
                            <instance_generator name="URNfromTextualContent">
				<arg name="text" type="xpath">text()</arg>
			    </instance_generator>
                            <label_generator name="Literal">
                                <arg name="text" type="xpath">text()</arg>
				<arg name="language" type="constant">en</arg>
                            </label_generator>
                        </entity>
                    </target_node>
                </range>
            </link>
        </mapping>
    </mappings>
</x3ml>
```

Produced RDF output:
```
<rdf:RDF
    xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
    xmlns:loc="http://localost/schema/"
    xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
    <loc:Man-Made_Object rdf:about="http://localhost/coin_1">
        <loc:has_country>
            <loc:Country rdf:about="URN:UUID:d6d2800a-f5d1-3768-9c47-3a5494246c2e">
                <rdfs:label xml:lang="en">Ancient Greece</rdfs:label>
            </loc:Country>
        </loc:has_country>
    </loc:Man-Made_Object>
    <loc:Man-Made_Object rdf:about="http://localhost/coin_2">
        <loc:has_country>
            <loc:Country rdf:about="URN:UUID:c921a09f-e9ed-3bf2-a810-c8d290e01691">
                <rdfs:label xml:lang="en">Ancient Rome</rdfs:label>
            </loc:Country>
        </loc:has_country>
    </loc:Man-Made_Object>
    <loc:Man-Made_Object rdf:about="http://localhost/coin_3">
        <loc:has_country rdf:resource="URN:UUID:d6d2800a-f5d1-3768-9c47-3a5494246c2e"/>
    </loc:Man-Made_Object>
</rdf:RDF>
```

**Comment by ymark** [comment]
The files for the above example are attached

[input.xml.txt](https://github.com/isl/x3ml/files/1044424/input.xml.txt)
[mappings.x3ml.txt](https://github.com/isl/x3ml/files/1044421/mappings.x3ml.txt)
[output.rdf.txt](https://github.com/isl/x3ml/files/1044422/output.rdf.txt)
[generator-policy.xml.txt](https://github.com/isl/x3ml/files/1044423/generator-policy.xml.txt)

**Comment by ymark** [comment]
This new functionality will be made available under version 1.8.0
